operators/ior_presets.py

```diff
@@ -0,0 +1,181 @@
+    _ior_values["polystyrene"] = ["Polystyrene", 1.52]
+    _ior_values["pyrex_glass"] = ["Pyrex Glass", 1.470]
+    _ior_values["quartz"] = ["Quartz", 1.458]
+    _ior_values["sapphire"] = ["Sapphire", 1.77]
+    _ior_values["silicone_oil_20C"] = ["Silicone Oil @ 20C", 1.4]
+    _ior_values["turpentine"] = ["Turpentine", 1.47]
+    _ior_values["vacuum"] = ["Vacuum", 1]
+    _ior_values["water_20C"] = ["Water @ 20C", 1.330]
+    _ior_values["water_ice"] = ["Water Ice", 1.31]
+
+    # A default value that will be assigned an integer index of 0 as required
+    # by EnumProperty when using a callback to obtain the list of items.
+    default_key = "water_20C"
+
+    def get_sorted_list(self, sort="name"):
+        # Make a deep copy of the ior_values dict
+        ior_values = copy.deepcopy(self._ior_values)
+        # Append a unique integer index to each tuple. The EnumProperty needs
+        # this to determine a default item (index 0) when using a callback.
+        # Failure to do so results in a random item becoming the default.
+        ior_values[self.default_key].append(0)
+        index = 1
+        for ior in ior_values:
+            if ior != self.default_key:
+                ior_values[ior].append(index)
+                index += 1
+        # Convert the dict to a sorted list of tuples
+        if sort == "value":
+            index = 1
+        else:
+            index = 0
+        preset_list = []
+        for item in sorted(ior_values.items(),
+                           key=lambda e: e[1][index]):
+            text = "{} ({:f})".format(item[1][0], item[1][1])
+            preset_list.append((item[0], text, text, item[1][2]))
+        return preset_list
+
+    @classmethod
+    def get_item(cls, key=default_key):
+        ret = None
+        if key not in cls._ior_values.keys():
+            print("Error: key '{}' not found in "
+                  "LuxCoreIORPresets::get_item()".format(key))
+        else:
+            ret = cls._ior_values[key]
+        return ret
+
+    @classmethod
+    def get_value(cls, key=default_key):
+        ret = None
+        if key not in cls._ior_values.keys():
+            print("Error: key '{}' not found in "
+                  "LuxCoreIORPresets::get_value()".format(key))
+        else:
+            ret = cls._ior_values[key][1]
+        return ret
+
+
+# It seems that Blender *Property classes do not support inheritance if
+# there are methods in the same class. The following classes
+# (LuxCoreIORPresetCommonProperties, LuxCoreIORPresetBase,
+#  LUXCORE_OT_ior_preset_names  and LUXCORE_OT_ior_preset_values) are an
+# attempt at minimizing code duplication within the limits imposed by Blender.
+class LuxCoreIORPresetCommonProperties():
+    """ A property-only class to work around Blender's inheritance limits. """
+    bl_idname = "luxcore.ior_preset_common_properties"
+    bl_label = ""
+    node_name = StringProperty()
+    node_tree_name = StringProperty()
```

Comment:
You have to access the node_tree by index, like here: https://github.com/LuxCoreRender/BlendLuxCore/blob/master/operators/world.py#L51
Because there can be multiple node trees with the same name (for example when you link a node tree into the .blend file).